src/ast/visitor.h

```diff
@@ -209,6 +209,11 @@ class Visitor {
     visitImpl(block.expr);
     return default_value();
   }
+  // A default visitor for Macro is not provided, as alone they are not
+  // valid code - they must be expanded in their callsites. Thus, Macro
+  // visitation should be specially handled by the macro expansion pass.
+  //
+  // R visit(Macro &macro) { .. }
```

Comment:
Please provide a default visitor that does nothing, and include this comment inside. Then you should have the visitor with `program.macros` uncommented out below.

If a visitor defines `visito(Macro &macro)` then it should visit them. We shouldn't expect them to re-define the `Program` level visitor. But the comment is perfectly fair that unless this is provided explicitly, the default visitor for macros will not descend into that expression.